src/java/org/apache/cassandra/cql3/statements/QualifiedStatement.java

```diff
@@ -81,13 +81,27 @@ public String toString()
 
     public static abstract class Composite extends QualifiedStatement
     {
+        private static final int RAW_TEXT_MAX_LENGTH = 4096;
+
+        private String txt;
+
         Composite()
         {
             super(null);
         }
 
         protected abstract Iterable<? extends QualifiedStatement> getStatements();
 
+        public void setRawText(String txt)
+        {
+            this.txt = txt != null ? txt.substring(0, Math.min(txt.length(), RAW_TEXT_MAX_LENGTH)) : null;
```

Comment:
nit: if we actually do exceed the max length, it might be a nice UX tweak to include a "blah blah..." (ellipsis) at the end just to make it clear there's more content (but we're truncating).